Mathlib/Data/Nat/Prime.lean

```diff
@@ -455,6 +455,21 @@ theorem exists_dvd_of_not_prime2 {n : ℕ} (n2 : 2 ≤ n) (np : ¬Prime n) :
     (not_prime_iff_minFac_lt n2).1 np⟩
 #align nat.exists_dvd_of_not_prime2 Nat.exists_dvd_of_not_prime2
 
+theorem not_prime_of_dvd (h : ∃ m, m ∣ n ∧ m ≠ 1 ∧ m ≠ n) : ¬Prime n := by
+  contrapose! h
+  simp only [← or_iff_not_imp_left]
+  exact h.eq_one_or_self_of_dvd
+
+theorem not_prime_of_dvd2 (h : ∃ m, m ∣ n ∧ 2 ≤ m ∧ m < n) : ¬Prime n := by
```

Comment:
I feel like `1 < m` is more natural than `2 ≤ m` (this would need to be changed on all the lemmas, including the original two).

Reply:
Would like to hear other's opinion. When I wrote this, there's already `theorem exists_dvd_of_not_prime2 {n : ℕ} (n2 : 2 ≤ n) (np : ¬Prime n) : ...`. I'm just mirroring existing code.